src/clr-angular/utils/drag-and-drop/draggable.ts

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright (c) 2016 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {
+    AfterViewInit,
+    ComponentFactory,
+    ComponentFactoryResolver,
+    ContentChild,
+    Directive,
+    ElementRef,
+    EventEmitter,
+    Inject,
+    Injector,
+    Input,
+    OnDestroy,
+    Output,
+    Renderer2,
+    ViewContainerRef
+} from "@angular/core";
+import {Subscription} from "rxjs/Subscription";
+
+import {DomAdapter} from "../../data/datagrid/render/dom-adapter";
+
+import {DraggableGhost} from "./draggable-ghost";
+import {DraggableHandle} from "./draggable-handle";
+import {IfDragged} from "./if-dragged";
+import {DragPosition} from "./interfaces/client-position";
+import {DraggableEvent} from "./interfaces/draggable-event";
+import {DragAndDropDispatcher} from "./providers/drag-and-drop-dispatcher";
+import {DragDispatcher} from "./providers/drag-dispatcher";
+import {CUSTOM_GHOST_STATE, CUSTOM_GHOST_STATE_PROVIDER, CustomGhostState} from "./providers/draggable-tracker";
+
+@Directive({selector: "[clrDraggable]", providers: [DragDispatcher, DomAdapter, CUSTOM_GHOST_STATE_PROVIDER]})
+export class Draggable implements AfterViewInit, OnDestroy {
+    constructor(private dragDispatcher: DragDispatcher, private dragAndDrop: DragAndDropDispatcher,
```

Comment:
Your variables are named inconsistently. One is named `dragDispatcher` and the other is named `dragAndDrop`. Both are providers and both end with `Dispatcher`

Reply:
Will address this.